File: Java/com/ockey/pcc/ProgramingContest03.java
package com.ockey.pcc;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class ProgramingContest03 {
	public static void main (String[] args) {
		int n = 10;
		int max_value = 0;
		int min_value = 0;
		List<Integer> x = Arrays.asList(2, 6, 7);
		List<String> max = Arrays.asList("Right", "Right", "Right");
		List<String> min = Arrays.asList("Right", "Right", "Right");
		if (n < 1 || n > 50) System.out.println("n: Domain error. Must be (1 <= n <= 50).");
		for(int obj : x) {
			if (obj < 1 || obj > n) System.out.println("x: Domain error. Must be (1 <= x <= " + n + ").");
		}
		for (int i = 0; i < x.size(); i++) {
			if (x.get(i) < (n - x.get(i))) {
				min.set(i, "Left");
				if (max_value < (n - x.get(i))){
					max_value = (n - x.get(i));
				}
				if (min_value == 0 ||  min_value < x.get(i)){
					min_value = x.get(i);
				}
			} else {
				max.set(i, "Left");
				if (max_value < x.get(i)){
					max_value = x.get(i);
				}
				if (min_value == 0 || min_value < (n - x.get(i))){
					min_value = (n - x.get(i));
				}
			}
		}
		PrintArray("Max  " + max_value + " : ", max);
		PrintArray("Min  " + min_value + " : ", min);
	}
	public static void PrintArray (String str, List<String> list) {
		System.out.print(str);
		for(String obj : list) {
			System.out.print(obj + ",");
		}
		System.out.println();
	}
}

File: Scala/ProgramingContest03.scala
import scala.collection.mutable.ListBuffer

object ProgramingContest03 {
	def main(args: Array[String]) {
		var n = 10
		var max_value = 0
		var min_value = 0
		var x = List(2, 6, 8)
		var max = ListBuffer("Right", "Right", "Right")
		var min = ListBuffer("Right", "Right", "Right")
		if (n < 1 || n > 50) System.out.println("n: Domain error. Must be (1 <= n <= 50).")
		for(obj <- x) {
			if (obj < 1 || obj > n) System.out.println("x: Domain error. Must be (1 <= x <= " + n + ").")
		}
		for (i <- 0 until x.size) {
			if (x(i) < (n - x(i))) {
				min.update(i, "Left")
				if (max_value < (n - x(i))){
					max_value = (n - x(i))
				}
				if (min_value == 0 ||  min_value < x(i)){
					min_value = x(i)
				}
			} else {
				max.update(i, "Left")
				if (max_value < x(i)){
					max_value = x(i)
				}
				if (min_value == 0 || min_value < (n - x(i))){
					min_value = (n - x(i))
				}
			}
		}
		PrintArray("Max  " + max_value + " : ", max)
		PrintArray("Min  " + min_value + " : ", min)
	}
	def PrintArray (str: String, list: ListBuffer[String]) {
		System.out.print(str)
		for(obj <- list) {
			System.out.print(obj + ",")
		}
		System.out.println()
	}
}
